de.tudarmstadt.ukp.wikipedia.api/src/main/java/de/tudarmstadt/ukp/wikipedia/api/hibernate/GenericDAO.java

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright 2017
+ * Ubiquitous Knowledge Processing (UKP) Lab
+ * Technische Universität Darmstadt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package de.tudarmstadt.ukp.wikipedia.api.hibernate;
+
+import de.tudarmstadt.ukp.wikipedia.api.Wikipedia;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.hibernate.LockOptions;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+
+/**
+ * A common base class for DAO classes.
+ * 
+ * @param <T> The entity type to provide persistence features for.
+ */
+public abstract class GenericDAO<T> {
+
+    private final Log logger = LogFactory.getLog(getClass());
+    
+    private final Wikipedia wiki;
+    private final SessionFactory sessionFactory;
+
+    private final String entityClass;
+
+    GenericDAO(Wikipedia wiki, String entityClass) {
+        this.wiki = wiki;
+        this.entityClass = entityClass;
+        this. sessionFactory = initializeSessionFactory();
+    }
+
+    private SessionFactory initializeSessionFactory() {
+        try {
+            return WikiHibernateUtil.getSessionFactory(wiki.getDatabaseConfiguration());
+        } catch (Exception e) {
+            logger.error("Could not locate SessionFactory in JNDI", e);
+            throw new IllegalStateException("Could not locate SessionFactory in JNDI");
```

Comment:
Exceptions should be chained. In the light of chaining, any particular reason why to log it here?

Reply:
That was an existing piece of code ("as is") taken from one of the DAO-classes in their original form. Shall I change this here, so `e` is chained into the new `IllegalStateException`?

Reply:
I think doing the chaining is a good idea. It would also be possible to open a separate cleanup PR and add this as one of the todos inside the issue.

Reply:
As this is a code smell and not a question of code style, I will fix the above occurrence with this next commit (pending). Please feel free to open up a new issue targeted to 1.2.1 milestone.